platform/error/platform-error-page-injector/src/main/java/org/codice/ddf/platform/error/injector/ErrorPageInjector.java

```diff
@@ -0,0 +1,96 @@
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.error.injector;
+
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.Map;
+import javax.servlet.ServletContext;
+import org.codice.ddf.platform.error.servlet.ErrorServlet;
+import org.eclipse.jetty.servlet.ErrorPageErrorHandler;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.hooks.service.EventListenerHook;
+import org.osgi.framework.hooks.service.ListenerHook.ListenerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ErrorPageInjector implements EventListenerHook {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ErrorPageInjector.class);
+
+  @Override
+  public void event(ServiceEvent event, Map<BundleContext, Collection<ListenerInfo>> listeners) {
+    if (event.getType() == ServiceEvent.REGISTERED) {
+      Bundle refBundle = event.getServiceReference().getBundle();
+      BundleContext bundlectx = refBundle.getBundleContext();
+      Object service = bundlectx.getService(event.getServiceReference());
+      if (service instanceof ServletContext) {
+        injectError((ServletContext) service, refBundle);
+      }
+    }
+  }
+
+  private void injectError(ServletContext context, Bundle refBundle) {
+
+    Field field = null;
+    try {
+      // this grabs the enclosing instance class, which is actually a private class
+      // this is the only way to do this in Java
+      field = context.getClass().getDeclaredField("this$0");
+      field.setAccessible(true);
+    } catch (NoSuchFieldException e) {
+      LOGGER.warn(
+          "Unable to find enclosing class of ServletContext for delegating the error page. The default jetty errors will display in the browser",
+          e);
+    }
+    ServletHandler handler = null;
+    if (field != null) {
+      // need to grab the servlet context handler so we can get down to the handler, which is what
+      // we really need
+      ServletContextHandler httpServiceContext = null;
+      try {
+        httpServiceContext = (ServletContextHandler) field.get(context);
+      } catch (IllegalAccessException e) {
+        LOGGER.warn(
+            "Unable to get the ServletContextHandler for {}. The default jetty errors may display to the browser.",
```

Comment:
✏️ I like this message, but it may be good to say something like `Jetty's default error page will be used for this context.` instead of `The default jetty errors may display to the browser.`